FFmpegInteropXWinUI/AbstractEffectFactory.h

```diff
@@ -0,0 +1,13 @@
+#pragma once
+#include "pch.h"
+#include "IAvEffect.h"
+
+namespace FFmpegInteropX {
+	class AbstractEffectFactory abstract
+	{
+	public:		
+		virtual std::shared_ptr<IAvEffect> CreateEffect(winrt::hstring definitions) abstract;
```

Comment:
Prevents direct class instanctation.

Reply:
ah now I realize this also is a leftover from c++/CX. Again, not sure why the compiler isn't issuing an error, I use the "C++/winRT windows runtime component tenplate" from visual studio to generate that project,

Goes to show just how much of a freak show this c++/winRT actually is. As I struggle to port stuff I wonder if this is really worth it and it isn't just a temporary tantrum from Microsoft.

Reply:
C++/WinRT is not a language dialect, nor a language extension.
It's a set of standard C++ generated projections for lower level Windows and COM APIs.

In my opinion, it's very worth it because of its standards compliance.
It also saves a lot of boilerplate code as opposed to using plain WINAPI.